src/frontend/src/pages/FinancePage/CreateReimbursementRequestForm/CreateReimbursementRequestForm.tsx

```diff
@@ -0,0 +1,188 @@
+import { useHistory } from 'react-router-dom';
+import { routes } from '../../../utils/routes';
+
+export interface CreateReimbursementRequestFormInput {
+  vendorId: string;
+  account: ClubAccount;
+  dateOfExpense: Date;
+  expenseTypeId: string;
+  reimbursementProducts: ReimbursementProductCreateArgs[];
+  receiptFiles: {
+    file: File;
+  }[];
+}
+
+const schema = yup.object().shape({
+  vendorId: yup.string().required('Vendor is required'),
+  account: yup.string().required('Account is required'),
+  dateOfExpense: yup.date().required('Date of Expense is required'),
+  expenseTypeId: yup.string().required('Expense Type is required'),
+  reimbursementProducts: yup
+    .array()
+    .of(
+      yup.object().shape({
+        wbsNum: yup.object().required('WBS Number is required'),
+        name: yup.string().required('Description is required'),
+        cost: yup.number().required('Amount is required').min(1, 'Amount must be greater than 0')
+      })
+    )
+    .required('reimbursement products required')
+    .min(1, 'At least one Reimbursement Product is required'),
+  receiptFiles: yup.array().required('receipt files required').min(1, 'At least one Receipt is required')
+});
+
+const CreateReimbursementRequestForm = () => {
+  const {
+    handleSubmit,
+    control,
+    formState: { errors }
+  } = useForm({
+    resolver: yupResolver(schema),
+    defaultValues: {
+      vendorId: '',
+      account: ClubAccount.CASH,
+      dateOfExpense: new Date(),
+      expenseTypeId: '',
+      reimbursementProducts: [] as ReimbursementProductCreateArgs[],
+      receiptFiles: [] as {
+        file: File;
+      }[]
+    }
+  });
+
+  const {
+    fields: receiptFiles,
+    append: receiptAppend,
+    remove: receiptRemove
+  } = useFieldArray({
+    control,
+    name: 'receiptFiles'
+  });
+  const {
+    fields: reimbursementProducts,
+    append: reimbursementProductAppend,
+    remove: reimbursementProductRemove
+  } = useFieldArray({
+    control,
+    name: 'reimbursementProducts'
+  });
+
+  const { isLoading: createReimbursementRequestIsLoading, mutateAsync: createReimbursementRequest } =
+    useCreateReimbursementRequest();
+  const { isLoading: uploadReceiptsIsLoading, mutateAsync: uploadReceipts } = useUploadManyReceipts();
+  const {
+    isLoading: allVendorsIsLoading,
+    isError: allVendorsIsError,
+    error: allVendorsError,
+    data: allVendors
+  } = useGetAllVendors();
+  const {
+    isLoading: allExpenseTypesIsLoading,
+    isError: allExpenseTypesIsError,
+    error: allExpenseTypesError,
+    data: allExpenseTypes
+  } = useGetAllExpenseTypes();
+
+  const {
+    isLoading: allProjectsIsLoading,
+    isError: allProjectsIsError,
+    error: allProjectsError,
+    data: allProjects
+  } = useAllProjects();
+
+  const toast = useToast();
+  const history = useHistory();
+  const totalCost = reimbursementProducts.reduce((acc, curr) => Number(acc) + Number(curr.cost), 0);
```

Comment:
why are you casting these as numbers? also make the variable names more descriptive

Reply:
there was something very weird where it would add them like strings so 0 + 10 = 010 instead of just 10 yk

Reply:
fair enough